kong/plugins/mashape-analytics/buffer.lua

```diff
@@ -26,15 +32,27 @@ local ngx_timer_at = ngx.timer.at
 local table_insert = table.insert
 local table_concat = table.concat
 local table_remove = table.remove
+local string_sub = string.sub
 local string_len = string.len
 local string_rep = string.rep
 local string_format = string.format
+local math_pow = math.pow
+local math_min = math.min
 local setmetatable = setmetatable
 
 local MB = 1024 * 1024
-local MAX_BUFFER_SIZE = 1 * MB
+local MAX_BUFFER_SIZE = 500 * MB
```

Comment:
Is this change from 1MB to 500MB correct? Seems awfully large. This is the maximum payload size that the socket server will accept in a single request, right? Given that `max_queue_size` defaults to 10MB, and the queue holds multiple batches, I would expect this value to be < 10MB.